src/libs/UserUtils.ts

```diff
@@ -68,38 +78,62 @@ function hashText(text: string, range: number): number {
     return Math.abs(hashCode(text.toLowerCase())) % range;
 }
 
+/**
+ * Generate a random accountID base on searchValue.
+ */
+function generateAccountID(searchValue: string): number {
+    return hashText(searchValue, 2 ** 32);
+}
+
 /**
  * Helper method to return the default avatar associated with the given accountID
  * @param [accountID]
  * @returns
  */
-function getDefaultAvatar(accountID = -1): React.FC<SvgProps> {
+function getDefaultAvatar(accountID = -1, avatarURL?: string): React.FC<SvgProps> {
     if (accountID <= 0) {
         return FallbackAvatar;
     }
     if (Number(accountID) === CONST.ACCOUNT_ID.CONCIERGE) {
         return ConciergeAvatar;
     }
-
     // There are 24 possible default avatars, so we choose which one this user has based
     // on a simple modulo operation of their login number. Note that Avatar count starts at 1.
-    const accountIDHashBucket = ((accountID % CONST.DEFAULT_AVATAR_COUNT) + 1) as AvatarRange;
-
+    let accountIDHashBucket: AvatarRange;
+    if( avatarURL && avatarURL.includes('images/avatars/default-avatar_')){
+        const lastDigitIndex = avatarURL.lastIndexOf('_');
```

Comment:
I wonder if it'd be a bit tidier (and ultimately safer!) to just use regex to extract the avatar ID?